components/LoginButton/PrivyButton.tsx

```diff
@@ -6,14 +6,24 @@ import truncateAddress from "@/lib/truncateAddress";
 import { usePrivy } from "@privy-io/react-auth";
 import { Address } from "viem";
 import Image from "next/image";
+import { useState, useEffect } from "react";
+import { DropdownMenu } from "./DropdownMenu";
 
 interface PrivyButtonProps {
   className?: string;
 }
+
 export function PrivyButton({ className = "" }: PrivyButtonProps) {
   const { login, ready, logout } = usePrivy();
   const { connectedWallet } = useConnectedWallet();
   const { data } = useArtistProfile(connectedWallet as Address);
+  const [isOpen, setIsOpen] = useState(false);
+
+  useEffect(() => {
+    if (connectedWallet) {
+      setIsOpen(false);
+    }
+  }, [connectedWallet]);
```

Comment:
what is the purpose of this useEffect?
- it seems like we `close` the popup when wallet is connected.
- Why? 
imo this code block should be `removed` 🚮